pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -1759,18 +1720,76 @@ func appendKubernetesRemoteLocation(xsite *ispnutil.XSite, infinispan *infinispa
 	return nil
 }
 
-func getRemoteSiteServiceHostPort(service *corev1.Service, remoteKubernetes *ispnutil.Kubernetes, logger logr.Logger) (string, int32, error) {
+func getCrossSiteServiceHostPort(service *corev1.Service, kubernetes *ispnutil.Kubernetes, logger logr.Logger) (string, int32, error) {
 	switch serviceType := service.Spec.Type; serviceType {
 	case corev1.ServiceTypeNodePort:
 		// If configuring NodePort, expect external IPs to be configured
-		return remoteKubernetes.PublicIP(), remoteKubernetes.GetNodePort(service), nil
+		return getNodePortServiceHostPort(service, kubernetes, logger)
 	case corev1.ServiceTypeLoadBalancer:
 		return getLoadBalancerServiceHostPort(service, logger)
 	default:
 		return "", 0, fmt.Errorf("unsupported service type '%v'", serviceType)
 	}
 }
 
+func getNodePortServiceHostPort(service *corev1.Service, k *ispnutil.Kubernetes, logger logr.Logger) (string, int32, error) {
+	//The IPs must be fetch. Some cases, the API server (which handles REST requests) isn't the same as the worker
+	//So, we get the workers list. It needs some permissions cluster-reader permission
+	//oc create clusterrolebinding <name> -n ${NAMESPACE} --clusterrole=cluster-reader --serviceaccount=${NAMESPACE}:<account-name>
+	workerList := &corev1.NodeList{}
+
+	//select workers only. AFAIK, only the workers have the proxy to the pods
+	labelSelector := labels.SelectorFromSet(map[string]string{"node-role.kubernetes.io/worker": ""})
+
+	listOps := &client.ListOptions{
+		LabelSelector: labelSelector,
+	}
+	err := k.Client.List(context.TODO(), workerList, listOps)
+	if err != nil {
+		return "", 0, err
+	}
+
+	for _, node := range workerList.Items {
+		//host := k.PublicIP() //returns REST API endpoint. not good.
+		//iterate over the all the nodes and return the first ready.
+		nodeStatus := node.Status
+		for _, nodeCondition := range nodeStatus.Conditions {
```

Comment:
Yes, good implementation, @pruivo 
Just FYI: fields selector can be more elegant solution for k8s...but not supported in all cases